src/hooks/useElements.ts

```diff
@@ -0,0 +1,56 @@
+import { useMemo, useCallback } from "react";
+import { Element } from "../Element";
+import { useLocale } from "./useLocale";
+
+const elementsMap = import.meta.globEager("../data/elements/*.json");
+
+export function useElements() {
+  const { i18n } = useLocale();
+
+  const elements = useMemo(() => Object.values(elementsMap) as Element[], []);
+  const elementLookup = useMemo(
+    () =>
+      Object.fromEntries(elements.map((element) => [element.atomic, element])),
+    [elements]
+  );
```

Comment:
This can be moved outside the hook as those won't change.

Reply:
@YPAzevedo can you make a PR for this if you have the time 😄